sites/example-project/src/components/ui/CodeBlock.svelte

```diff
@@ -0,0 +1,121 @@
+<script>
+  export let source;
+  export let copyToClipboard = true;
+  import Copy from "./Deployment/CopyIcon.svelte";
+  import Success from "./Deployment/CopySuccessIcon.svelte";
+  let copied = false;
+
+  const toggleCopied = function () {
+    copied = false;
+  };
+
+  export let copy = async (text) => {
+    try {
+      if (!copied) {
+        await navigator.clipboard.writeText(text);
+        copied = true;
+        setTimeout(toggleCopied, 1500);
+      }
+    } catch (e) {}
+  };
+</script>
+
+<pre><code>{#if source}{source}{:else}<slot />{/if}</code>
+{#if copyToClipboard}
+<button
+  type="button"
+  class="container"
+  class:copied
+  on:click={() => {
+    if (source !== undefined) {
+      copy(source);
+    }
+  }}>{#if copied}<Success color=var(--green-500)/>{:else}<Copy/>{/if}</button>
```

Comment:
It's better to give  `if/else` indentations and spaces

```svelte
    {#if copied}
      <Success color=var(--green-500)/>
    {:else}
      <Copy/>
    {/if}

```

Reply:
I am not sure why, but when I included the indentations, it introduced whitespace into the component near the Copy Icon. This was the workaround i found to remove it.

Reply:
This is some kind of svelte bug / weird behavior: https://github.com/sveltejs/svelte/issues/3204

I've edited to format it as readably as possible, but adding indents causes them to show up as whitespace in the codeblock.